File: Data Structures & Algorithms/string-encode-and-decode/submission-3.js
class Solution {
    /**
     * @param {string[]} strs
     * @returns {string}
     */
    encode(strs) {
      const count = String(strs.length);
      strs.unshift(count);
      return strs.join('01010101010101010001');
    }

    /**
     * @param {string} str
     * @returns {string[]}
     */
    decode(str) {
      const ans = str.split('01010101010101010001')
      const count = Number(ans[0]);
      if(count===0){
        return [];
      }
     ans.shift();
     return ans;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-3.cpp
class Solution {
private:
unordered_map<int,int> dp;
public:
    Solution(){
      dp[0]=1;
      dp[1]=1;
      dp[2]=2;
      dp[3]=3;
    }
    int climbStairs(int n) {

        vector<int> ans(n+2,0);
        ans[0]=1;
        for(int i=0;i<n;i++){
          ans[i+1]+=ans[i];
          ans[i+2]+=ans[i];
        }
        return ans[n];
        // if(n<0){
        //   return 0;
        // }
        // if(dp[n]){
        //   return dp[n];
        // }

        // if(!dp.contains(n-1)){
        //   dp[n-1] = climbStairs(n-1);
        // }
        // if(!dp.contains(n-2)){
        //   dp[n-2] = climbStairs(n-2);
        // }

        // return dp[n-1]+dp[n-2];
    }
};
